### PR TITLE
:recycle: Pass objects by reference

### PR DESCRIPTION
**Pull request by canassa**
- Passing objects by reference
- Renamed a few functions

### CODE REVIEW

File: Source/objects.cpp [hunk 1]
```diff
@@ -1067,33 +1067,32 @@ void InitializeMicroDoor(Object &door)
 	ObjSetMicro(door.position, pieceNumber);
 }
 
-void AddSarc(int i)
+void AddSarcophagus(Object &sarcophagus)
 {
-	dObject[Objects[i].position.x][Objects[i].position.y - 1] = -(i + 1);
```
Comment (ephphatha) [inline Source/objects.cpp:1072]: This is an initialisation function that marks the tiles occupied by a large object. Sarcophagi occupy two tiles, this needs to be kept in. This is breaking the demo test.

File: Source/objects.cpp [hunk 2]
```diff
@@ -1164,140 +1163,140 @@ void AddShrine(int i)
...
 
-void AddCauldron(int i)
+void AddCauldron(Object &cauldron)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
+	cauldron._oRndSeed = AdvanceRndSeed();
 }
 
-void AddMurkyFountain(int i)
+void AddMurkyFountain(Object &murkyFountain)
```
Comment (ephphatha) [inline Source/objects.cpp:1219]: Have a look at #4571, a lot of these functions can be combined.

File: Source/objects.cpp (hunk 1 above)
Comment (canassa) [reply Source/objects.cpp:1072]: That was removed by mistake, let me put it back

Comment (canassa) [reply Source/objects.cpp:1072]: Done

File: Source/objects.cpp (hunk 2 above)
Comment (canassa) [reply Source/objects.cpp]: Done

File: Source/objects.cpp [hunk 3]
```diff
@@ -1164,140 +1164,130 @@ void AddShrine(int i)
...
+void AddBookcase(Object &bookcase)
 {
-	Objects[i]._oRndSeed = AdvanceRndSeed();
-	Objects[i]._oPreFlag = true;
+	bookcase._oRndSeed = AdvanceRndSeed();
+	bookcase._oPreFlag = true;
 }
 
-void AddBookstand(int i)
+void AddBookstand(Object &bookstand)
```
Comment (ephphatha) [inline Source/objects.cpp:1180]: The other big cleanup that would be worth including from #4571 is combining all these functions which only set `Object::_oRndSeed` :D

**Review by glebm — APPROVED** [review]

**Pull request merged**